components/fxa-client/src/internal/migrator.rs

```diff
@@ -2,41 +2,13 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-use crate::{error::*, scoped_keys::ScopedKey, scopes, FirefoxAccount};
-use ffi_support::IntoFfi;
+pub use crate::{FxAMigrationResult, MigrationState};
+
+use super::{error::*, scoped_keys::ScopedKey, scopes, FirefoxAccount};
 use serde_derive::*;
+use std::convert::TryFrom;
 use std::time::Instant;
 
-// Values to pass back to calling code over the FFI.
-
-#[derive(Serialize, Deserialize, PartialEq, Debug, Clone, Default)]
-pub struct FxAMigrationResult {
-    pub total_duration: u128,
```

Comment:
I'm pretty sure the recipients of this field were not handling it as an unsigned 128-bit integer; I've made it an i64 in the new bindings.